File: src/agents/Support/pyramid_analyst.py
import os

class PyramidAnalyst:
    """Assistente Técnico: Auditor de Distribuição de Testes (Pirâmide) 📐"""
    
    def analyze(self, map_data: dict, read_func) -> dict:
        """Classifica os testes entre Unit, Integration e E2E."""
        pyramid = {"unit": 0, "integration": 0, "e2e": 0, "total": 0}
        
        for file in map_data.keys():
            if "tests/" not in file.replace(os.sep, "/"): continue
            
            content = read_func(file)
            if not content: continue
            
            pyramid["total"] += 1
            if "mock" in content or "patch" in content: pyramid["integration"] += 1
            elif "selenium" in content or "integration_test" in content: pyramid["e2e"] += 1
            else: pyramid["unit"] += 1
        
        return pyramid

File: src_local/agents/Support/pyramid_analyst.ts
import winston from 'winston';

/**
 * Assistente Técnico: Auditor de Distribuição de Testes (Pirâmide) 📐
 */
export class PyramidAnalyst {
    /**
     * Classifica os testes entre Unit, Integration e E2E.
     */
    async analyze(mapData: Record<string, any>, readFunc: (path: string) => Promise<string | null>): Promise<any> {
        winston.debug("Analysing test pyramid distribution...");
        const pyramid = { unit: 0, integration: 0, e2e: 0, total: 0 };

        for (const file of Object.keys(mapData)) {
            if (!file.includes("tests/")) continue;

            const content = await readFunc(file);
            if (!content) continue;

            pyramid.total += 1;
            if (content.includes("mock") || content.includes("patch")) {
                pyramid.integration += 1;
            } else if (content.includes("selenium") || content.includes("integration_test")) {
                pyramid.e2e += 1;
            } else {
                pyramid.unit += 1;
            }
        }

        return pyramid;
    }
}
